### DISCUSSION TITLE
Garbled characters may appear when uploading Chinese files.

### DISCUSSION THREAD

**Discussion by clhome** [payloadcms/payload#2628] [Q&A] [answered]
When I upload an image and the file name is in Chinese, garbled characters appear on the webpage. How can I solve this problem?
this is the[ image link](https://www.jamesqq.top/img/other/%E6%B0%94%E7%90%83.png)
![chinesepic](https://www.jamesqq.top/img/other/%E6%B0%94%E7%90%83.png)
![Snipaste2](https://www.jamesqq.top/img/other/Snipaste2.png)
![Snipaste](https://www.jamesqq.top/img/other/Snipaste.png)
I searched online and found that if the file name appears garbled when uploading using Node FormData, it may be because the file name is not encoded in UTF-8. We need to use the encodeURI() function or other encoding methods to solve this problem.like this:
```typescript
const form = new FormData();
form.append('file', fs.createReadStream(filePath), encodeURI(fileName));
```
However, I don’t know where to place these codes. Does anyone know?

**Comment by JessRynkar** [comment]
Hey @clhome, I'm looking into this now 🔎

**Comment by JessRynkar** [comment, marked as answer]
To fix this issue, add the following property to the `upload` object in your `payload.config`: 

```ts
upload: {
    defParamCharset: 'utf-8',
},
```

Here is a little background as to why... 

Payload uses the [express-fileupload](https://www.npmjs.com/package/express-fileupload) package to manage uploads. By default, this package has `defParamCharset` set to `latin1`.  However, we need a character set that supports Chinese i.e. `utf-8`.

You have full control over the `express-fileupload` options with Payload, simply pass any [valid properties](https://github.com/mscdex/busboy#api) to the upload object in your base `payload.config`.

Note: here is the `defParamCharset` definition from `express-fileupload`:
<img width="841" alt="Screen Shot 2023-05-10 at 11 19 56 AM" src="https://github.com/payloadcms/payload/assets/67977755/fb68ab6f-4dc7-4a08-930c-567aadffea1d">

**Reply by clhome** [reply]
Thank you very much, the problem is solved, this problem has been bothering me for a long time.

**Reply by RiceeeChang** [reply]
[10:29:56] INFO (payload): Starting Payload...
[10:29:57] ERROR (payload): There were 1 errors validating your Payload config
[10:29:57] ERROR (payload): 1: Collection "media" > "upload.defParamCharset" is not allowed

Why apear these error messages, after I change the config?

**Reply by JessRynkar** [reply]
Hi @RiceeeChang - you don't need to add `defParamCharset` to your uploads **collection** config, instead it needs to be added to your **_payload.config.ts_**. If you remove it from your media collection, and add it into the payload.config, the error should go away.

**Reply by RiceeeChang** [reply]
thanks, 

The problem is solved, and the fileupload with chinese filename is working well.
